pkg/3scale/amp/operator/backend_reconciler.go

```diff
@@ -154,6 +154,60 @@ func (r *BackendReconciler) Reconcile() (reconcile.Result, error) {
 	if err != nil {
 		return reconcile.Result{}, err
 	}
+	if r.apiManager.Spec.Backend.ListenerSpec.Hpa.Enabled || r.apiManager.Spec.Backend.WorkerSpec.Hpa.Enabled {
+		if redisStorageUrl != redisQueuesUrl {
+			hpa := r.apiManager.Spec.Backend.ListenerSpec.Hpa
+			// If any of the ApiManager hpa fields are populated then you are in a sync with the ApiManger values
+			if hpa.MinPods != nil || hpa.MaxPods != 0 || hpa.CpuPercent != nil || hpa.MemoryPercent != nil {
+				err = r.ReconcileHpa(backend.BackendListenerHpa(r.apiManager.Namespace, hpa.MinPods, hpa.MaxPods, hpa.CpuPercent, hpa.MemoryPercent), reconcilers.GenericHPAMutator)
+				if err != nil {
+					return reconcile.Result{}, err
+				}
+			} else {
+				err = r.ReconcileHpa(backend.BackendListenerHpa(r.apiManager.Namespace, hpa.MinPods, hpa.MaxPods, hpa.CpuPercent, hpa.MemoryPercent), reconcilers.CreateOnlyMutator)
+				if err != nil {
+					return reconcile.Result{}, err
+				}
+			}
+			hpa = r.apiManager.Spec.Backend.WorkerSpec.Hpa
+			// If any of the ApiManager hpa fields are populated then you are in a sync with the ApiManger values
+			if hpa.MinPods != nil || hpa.MaxPods != 0 || hpa.CpuPercent != nil || hpa.MemoryPercent != nil {
+				err = r.ReconcileHpa(backend.BackendWorkerHpa(r.apiManager.Namespace, hpa.MinPods, hpa.MaxPods, hpa.CpuPercent, hpa.MemoryPercent), reconcilers.GenericHPAMutator)
+				if err != nil {
+					return reconcile.Result{}, err
+				}
+			} else {
+				err = r.ReconcileHpa(backend.BackendWorkerHpa(r.apiManager.Namespace, hpa.MinPods, hpa.MaxPods, hpa.CpuPercent, hpa.MemoryPercent), reconcilers.CreateOnlyMutator)
+				if err != nil {
+					return reconcile.Result{}, err
+				}
+			}
+		} else {
+			// set status message if logical redis db are detected in the backend
+			message := "logical redis instances found in the backend, which is blocking redis async mode, horizontal pod autoscaling for backend cannot be enabled without async mode"
+			r.logger.Info(message)
+			if r.apiManager.Status.Conditions != nil {
+				if r.apiManager.Status.Conditions.IsTrueFor("Available") {
+					if r.apiManager.Status.Conditions[0].Message == "" {
+						r.apiManager.Status.Conditions[0].Message = message
+						r.UpdateResourceStatus(r.apiManager)
+					}
+				}
+			}
+
+		}
+	} else {
```

Comment:
I think we should improve this a bit. 
Either now or in the future, we might want to log a message to APIM unrelated to HPA entirely. Are we at risk then that the message will be cleared every time by the code below if not running in HPA mode for backends?

Reply:
agree had some taughts on it here https://issues.redhat.com/browse/THREESCALE-10563?focusedId=23610601&page=com.atlassian.jira.plugin.system.issuetabpanels:comment-tabpanel#comment-23610601

Reply:
removing status updates from the pr replace with log messages until we deside a way forward.

Reply:
done